### PR TITLE
VladDmutriv - create new framework for RozetkaAutomation

### PR DESCRIPTION
**Pull request by vladdmytriv**
another type of framework for Rozetka 

I saw that we will have a lot of same PO's and methods in it on Rozetka branch. Lots of element open the same page so I try to identify needed pages and write methods that will open the same page from different elements.

### CODE REVIEW

File: src/main/java/com/ss/ita/google/ui/pages/HomePage.java [hunk 1]
```diff
@@ -2,9 +2,8 @@
 
 import org.openqa.selenium.WebElement;
 
-import static com.codeborne.selenide.Selectors.byName;
```
Comment (vladdmytriv) [inline src/main/java/com/ss/ita/google/ui/pages/HomePage.java:5]: just fix conflicts after merging

File: src/main/java/com/ss/ita/rozetka/ui/Modals/BasketModal.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import static com.codeborne.selenide.Selenide.$x;
+
+public class BasketModal {
+
+    public OrderingPage openOrderingPage() {
+        $x("//a[contains(@class, 'cart-receipt__submit')]").click();
+        return new OrderingPage();
+    }
+
+    public String getPriceWithCurrency() {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/Modals/BasketModal.java:16]: what if more than one product?

File: src/main/java/com/ss/ita/rozetka/ui/Modals/BasketModal.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        return $x("//div[contains(@class,'sum-price')]").getAttribute("textContent");
+    }
+
+    public int getPrice() {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/Modals/BasketModal.java:20]: same thing

File: src/main/java/com/ss/ita/rozetka/ui/ProductsEnum/GeneralProductCategories.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.ss.ita.rozetka.ui.ProductsEnum;
+
+public enum GeneralProductCategories {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/ProductsEnum/GeneralProductCategories.java:3]: singular

File: src/main/java/com/ss/ita/rozetka/ui/ProductsEnum/ProductCategoriesAndSubCategories.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package com.ss.ita.rozetka.ui.ProductsEnum;
+
+public enum ProductCategoriesAndSubCategories {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/ProductsEnum/ProductCategoriesAndSubCategories.java:3]: same

File: src/main/java/com/ss/ita/rozetka/ui/pages/HomePage.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import com.codeborne.selenide.SelenideElement;
+
+import static com.codeborne.selenide.Selenide.*;
+import static java.lang.String.format;
+
+public class HomePage extends Header{
+
+    private static final String GENERAL_PRODUCT_CATEGORY_TEMPLATE = "//div[contains(@class,'menu-wrapper_state_static')]/descendant::a[contains(@href,'%s')]";
+
+    public List<SelenideElement> getHomePageProductList(){
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/HomePage.java:14]: no selenide elems sticking out of PO

File: src/test/java/com/ss/ita/rozetka/verifyOpeningProductPage.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+public class verifyOpeningProductPage extends TestRunnerForRozetka {
+
+    @Test
+    public void verifyOpenProductPage() {
+
+        new HomePage()
+                .openProductCategoryPage(COTTAGE_GARDEN_BACKYARD.getName())
+                .openProductTypePage(GARDEN_TECH_CATEGORY.getName())
+                .openProductPage(1);
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/verifyOpeningProductPage.java:18]: this not complete, do not create not finished PRs

File: src/test/java/com/ss/ita/rozetka/verifyOpeningProductPage.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
+package com.ss.ita.rozetka;
+
+import com.ss.ita.rozetka.ui.pages.*;
+import com.ss.ita.rozetka.ui.runner.TestRunnerForRozetka;
+import org.testng.annotations.Test;
+
+import static com.ss.ita.rozetka.ui.ProductsEnum.GeneralProductCategories.*;
+import static com.ss.ita.rozetka.ui.ProductsEnum.ProductCategoriesAndSubCategories.*;
+
+public class verifyOpeningProductPage extends TestRunnerForRozetka {
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/verifyOpeningProductPage.java:10]: ???

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/ita/rozetka/ui/pages/HomePage.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.codeborne.selenide.Selenide;
+
+import static com.codeborne.selenide.Selenide.$x;
+import static com.codeborne.selenide.Selenide.open;
+import static java.lang.String.format;
+
+public class HomePage extends Header {
+
+    private static final String GENERAL_PRODUCT_CATEGORY_TEMPLATE = "//div[contains(@class,'menu-wrapper_state_static')]/descendant::a[contains(@href,'%s')]";
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/HomePage.java:11]: if used once, inline

File: src/main/java/com/ss/ita/rozetka/ui/pages/ProductCategoryPage.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
+package com.ss.ita.rozetka.ui.pages;
+
+import static com.codeborne.selenide.Selenide.*;
+import static java.lang.String.format;
+
+public class ProductCategoryPage extends Header {
+
+    private static final String PRODUCT_CATEGORY_OR_SUBCATEGORY_TEMPLATE = "//a[contains(@href,'%s')]";
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/ProductCategoryPage.java:8]: same

Comment (toleksyn) [reply src/main/java/com/ss/ita/rozetka/ui/pages/ProductCategoryPage.java:8]: same for all

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/ita/rozetka/ui/Modals/HamburgerModal.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+import com.codeborne.selenide.Condition;
+import com.ss.ita.rozetka.ui.pages.*;
+
+import static com.codeborne.selenide.Selenide.*;
+
+public class HamburgerModal {
+
+    public Header closeHamburgerModal() {
+        $x("//button[@class='side-menu__close']").click();
+        $x("//nav[@class='drawer ng-star-inserted']").shouldNotBe(Condition.visible);
```
Comment (vladdmytriv) [inline src/main/java/com/ss/ita/rozetka/ui/Modals/HamburgerModal.java:12]: Taras why U push your commits in my PR?

Like I know we use testng assertion, don't we?

File: src/main/java/com/ss/ita/rozetka/ui/Modals/CatalogModal.java [hunk 12]
```diff
@@ -0,0 +1,7 @@
+package com.ss.ita.rozetka.ui.Modals;
+
+import com.ss.ita.rozetka.ui.pages.*;
+
+public class CatalogModal extends Header {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/Modals/CatalogModal.java:5]: abstrtact HeaderPage {
      protected Header  header = new Header();
}

catalogModal.header.search()

**Pull request merged**